server/data/data-library/src/main/java/org/apache/james/domainlist/lib/AbstractDomainList.java

```diff
@@ -69,8 +69,12 @@ public void configure(HierarchicalConfiguration config) throws ConfigurationExce
     }
 
     @Override
-    public String getDefaultDomain() {
-        return defaultDomain;
+    public String getDefaultDomain() throws DomainListException {
```

Comment:
Nope.
I'm saying if it's null we had a problem with configuration. Rather than returning a null result, let's say to the world we had a problem.
And hopefully RemoteDelivery handles DomainList exception and already backs up to "localhost" if a DomainList exception is encountered. It means sender threads do not die stupidly anymore.
I think solving the initialization problem fully and cleanly is out of the scope of this PR and should require some more work.
This temporary work around makes sens to me, and solves the issue temporary.
And no it do not fail anymore.